vault/ha.go

```diff
@@ -384,6 +384,9 @@ func (c *Core) StepDown(httpCtx context.Context, req *logical.Request) (retErr e
 }
 
 func (c *Core) stopStandby() {
+	c.stateLock.Lock()
```

Comment:
If the `standbyStopCh` is atomic, why do we need `stateLock` here? It seems both redundant and latency-inducing. 

Should we instead split this into two?

```go
stopCh := c.standbyStopCh.Load()

if stopCh == nil {
  return 
}

select {
  ...
}
```